manila/share/drivers/netapp/dataontap/client/client_cmode.py

```diff
@@ -1859,12 +1859,32 @@ def modify_active_directory_security_service(
                 msg = _("Failed to modify existing CIFS server user-name. %s")
                 raise exception.NetAppException(msg % e.message)
 
+        if 'defaultadsite' in differring_keys:
+            if new_security_service['defaultadsite'] is not None:
```

Comment:
yes, why not ? if current_security_service is not none, then based on whether new ss has server or not, accordingly cifs options will be configured.

Reply:
based on existing condition, if both are none
1. if old ss has default_ad_site, it should configure cifs options  <-- fixed now. 
2. if old ss has server, removed preferred dc will remove server.

Reply:
I lost track of your changes and got confused. Please do not force push during review.
Can you please summaries what the possible scenarios are in terms of the changes in server and default-ad-site?

Reply:
I went over case by case, and took some notes
       I think the list is too long for here, maybe move discussions to slack, or have a meeting
       
        # update defaultadsite:
        #   1862 -> 1863 ->
        #               are user/password always available from request? what if they are not provided
        #               if they are provided, then does it mean that L1840 and L1851 are true?
        #               is it necessary to call configure_cifs_options? the option are not changed
        #
        # update server:
        #   1884 -> 1885 ->
        #        -> 1888 ->
        #              is it necessary call remove_preferred_dcs then set_preferred_dc?
        #              is it necessary to call configure_cifs_options? the option are not changed, right?
        #
        # change from server -> defaultadsite
        #   old server is not None, new server is None
        #   old defaultadsite is None, new defaultadsite is not None
        #   1862 -> 1863 -> ok
        #   1884 -> 1885 -> ok
        #
        # change from defaultadsite to server
        #  old server is None, new server is not None
        #  old defaultadsite is not None, new defaultadsite is None
        #  1862 -> 1880 ->
        #             but no code will be executed because of L1881
        #  1884 -> 1888 -> ok
        #             anything need to be done to revert the old defaultadsite settings?
        #             like the api_ags set between L1865 -> L1870
        #
        #  remove server
        #  1884 -> 1885 ->
        #             configure cifs options not needed?
        #
        #  remove defaultadsite
        #  1862 -> 1863 ->
        #             won't work if AD discovery mode is 'site' -> see ntapp kb
        #             configure_cifs_options should be called first
        #
        #   when will L1882 be visisted?

Reply:
1 - yes,added check
2- update server. yes we need to remove dc and add new dc. Also call configure cifs options after set new dc.
3- ok
4 - since server in dff keys, new dc and configure cifs options wil be triggerd. 
5,6 - remove server or remove defaultadsite case are not there in update as for CIFS either of them is must.